tests/integration/istioio/trafficmanagement/faultinjection/fault_injection_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faultinjection
```

Comment:
Can you make it in the toplevel trafficmanagement package. Each new package will be a fresh install of Istio which slows down the test. You would only need a new package if you need a custom install steps

Reply:
@howardjohn  I moved the test to toplevel package. As the configs on this task are very simple, i think we need this in order to generate the snipets for istio.io docs. lets see @brian-avery comments on this

Reply:
@howardjohn @elieser1101 These tests are supposed to be complete tests of the corresponding doc on isitio.io. This one is currently only doing a small subset of what the [fault injection task](https://istio.io/docs/tasks/traffic-management/fault-injection/) does (i.e., apply and verify the config is created). I'm assuming that the rest of the test is still TBD?

That said, I'm still convinced that we are writing these test with snippet generation in the wrong direction. We should generate the snippets from the doc, and use them to write the test, instead of the other way around.

**Proposal**: snippets should work in the opposite direction. Source of truth (snippets) should be the doc, not the test. We want to test the docs, not document the tests.

1. Doc author simply writes the doc with ordinary text sections for commands and output. Like today, more or less.
2. Test author annotates the doc to include annotations that identify text fragments as named snippets that can be used in a test program. From the annotations, we generate snippets of code to be used in the test program.
3. Test author writes a test that uses the generated snippets for the commands and output checking.

This approach leaves the commands in the doc and uses them in a “doc verification test” instead of trying to generate the docs from test programs written in advance. It also properly separates roles. Doc writers can spend their time doing what they do well, explaining how something works, not writing test programs.

 I described the general outline for this task (fault injection) here: https://github.com/istio/istio/pull/19606#discussion_r362642945